jas/runner/jasrunner.go

```diff
@@ -12,95 +12,111 @@ import (
-				return
-			}
-			if runAllScanners {
-				if configProfile == nil {
-					if len(scansToPreform) > 0 && !slices.Contains(scansToPreform, utils.IacScan) {
-						log.Debug("Skipping Iac scan as requested by input...")
-					} else if err = addModuleJasScanTask(module, jasutils.IaC, securityParallelRunner, runIacScan(securityParallelRunner, scanner, scanResults.ExtendedScanResults, module), errHandlerFunc); err != nil {
-						return
-					}
-				}
-				if len(scansToPreform) > 0 && !slices.Contains(scansToPreform, utils.SastScan) {
-					log.Debug("Skipping Sast scan as requested by input...")
-				} else if configProfile != nil {
-					log.Debug(fmt.Sprintf("Using config profile '%s' to determine whether to run Sast scan...", configProfile.ProfileName))
-					if configProfile.Modules[0].ScanConfig.SastScannerConfig.EnableSastScan {
-						err = addModuleJasScanTask(jfrogappsconfig.Module{}, jasutils.Sast, securityParallelRunner, runSastScan(securityParallelRunner, scanner, scanResults.ExtendedScanResults, module), errHandlerFunc)
-					} else {
-						log.Debug(fmt.Sprintf("Skipping Sast scan as requested by '%s' config profile...", configProfile.ProfileName))
-					}
-				} else if err = addModuleJasScanTask(module, jasutils.Sast, securityParallelRunner, runSastScan(securityParallelRunner, scanner, scanResults.ExtendedScanResults, module), errHandlerFunc); err != nil {
-					return
-				}
-			}
-		}
+	if err = addJasScanTaskForModuleIfNeeded(params, utils.ContextualAnalysisScan, runContextualScan(params.Runner, params.Scanner, params.ScanResults, params.Module, params.DirectDependencies, params.ThirdPartyApplicabilityScan, params.ApplicableScanType)); err != nil {
+		return
 	}
-
-	if configProfile != nil {
-		log.Debug("Config profile is in use. Skipping Contextual Analysis scan as it is not currently supported with a config profile...")
+	if params.ThirdPartyApplicabilityScan {
+		// Don't execute other scanners when scanning third party dependencies.
 		return
 	}
+	if err = addJasScanTaskForModuleIfNeeded(params, utils.SecretsScan, runSecretsScan(params.Runner, params.Scanner, params.ScanResults.JasResults, params.Module, params.SecretsScanType)); err != nil {
+		return
+	}
+	if !runAllScanners {
+		return
+	}
+	if err = addJasScanTaskForModuleIfNeeded(params, utils.IacScan, runIacScan(params.Runner, params.Scanner, params.ScanResults.JasResults, params.Module)); err != nil {
+		return
+	}
+	return addJasScanTaskForModuleIfNeeded(params, utils.SastScan, runSastScan(params.Runner, params.Scanner, params.ScanResults.JasResults, params.Module))
+}
 
-	if len(scansToPreform) > 0 && !slices.Contains(scansToPreform, utils.ContextualAnalysisScan) {
-		log.Debug("Skipping contextual analysis scan as requested by input...")
-		return err
+func addJasScanTaskForModuleIfNeeded(params JasRunnerParams, subScan utils.SubScanType, task parallel.TaskFunc) (err error) {
```

Comment:
I love how you re-wrote it, but we said this is the prioritization:
1) flags
2) config profile
3) jfrog apps config

it seems like we ignore the flags here that we can get from audit. Lets discuss it

Reply:
we are not ignoring the flags. you can see the first check is by flags:
```
if len(params.ScansToPreform) > 0 && !slices.Contains(params.ScansToPreform, subScan) {
	log.Debug(fmt.Sprintf("Skipping %s scan as requested by input...", subScan))
}
```
the second is by configProfile (has its own call to `addModuleJasScanTask` if exists to not continue to the last check) and only at the end we have the `jas.ShouldSkipScanner(params.Module, jasType)` check.